src/main/java/sopt/makers/authentication/application/service/user/UpdateUserProfileService.java

```diff
@@ -30,26 +31,45 @@ public class UpdateUserProfileService implements UpdateUserProfileUsecase {
   private final UserRepository userRepository;
   private final UserActivityHistoryRepository userActivityHistoryRepository;
   private final PhoneVerificationValidator phoneVerificationValidator;
+  private final UserCacheRepository userCacheRepository;
 
   @Transactional
   @Override
   public void updateUserProfile(UserProfileCommand command) {
-    User user = userRepository.findById(command.userId());
-    if (!user.getProfile().phone().equals(command.phone())) {
+    User user = userRepository.findByIdWithHistories(command.userId());
+    validatePhoneIfChanged(user, command);
+
+    updateUser(user, command);
+    updateUserActivities(user, command);
+    updateUserCache(user.getId());
+  }
+
+  private void validatePhoneIfChanged(User user, UserProfileCommand command) {
+    boolean isPhoneChanged = !user.getProfile().phone().equals(command.phone());
+
+    if (isPhoneChanged) {
       phoneVerificationValidator.validate(
           user.getProfile().name(), command.phone(), PhoneVerificationType.CHANGE_PHONE_NUMBER);
     }
-    ActivityList activityList = userActivityHistoryRepository.findByUser(user.getId());
+  }
+
+  private void updateUser(User user, UserProfileCommand command) {
     Profile updatedProfile =
         user.getProfile()
             .updateProfile(
-                command.email(), command.phone(),
-                command.birthday(), command.profileImage());
-    Map<Long, Activity> existingActivitiesById = findExistsActivities(activityList);
-    List<Activity> updatedActivities = updateActivities(command, existingActivitiesById);
+                command.email(), command.phone(), command.birthday(), command.profileImage());
+    userRepository.update(user, updatedProfile);
+  }
 
+  private void updateUserActivities(User user, UserProfileCommand command) {
+    Map<Long, Activity> existingActivitiesById = findExistsActivities(user.getActivities());
+    List<Activity> updatedActivities = updateActivities(command, existingActivitiesById);
     userActivityHistoryRepository.update(user, ActivityList.of(updatedActivities));
-    userRepository.update(user, updatedProfile);
+  }
+
+  private void updateUserCache(Long userId) {
+    User updatedUser = userRepository.findByIdWithHistories(userId);
+    userCacheRepository.put(updatedUser.getId(), updatedUser);
```

Comment:
```suggestion
  private void updateUserCache(User user, Profile updatedProfile, ActivityList updatedActivities) {
    User updatedUser = User.createUser(user.id, null, updatedProfile, updatedActivities)
    userCacheRepository.put(updatedUser.getId(), updatedUser);
```

em.flush(), em.clear() 안하고 updateUser(), updateUserActivities() 반환값으로 각각 업데이트 된 profile, activity 정보로 User 객체 만들어서 캐시에 저장할 수도 있을 것 같아요! (제가 이해를 제대로 한게 맞는지는 모르겠네요..)

Reply:
createUser를 사용할 생각을 못했네요!! 코드래빗도 동일하게 리뷰 남겨줘서 반영했습니다 감사해요 !!